User product guide (#2179)
ibexa/documentation-developer: docs/users/user_management_guide.md

---
description: The user management product guide provides a full description of its features as well as the benefits it brings to the client.
---

# User management product guide

User management is a fundamental aspect of any system.
[[= product_name =]] offers a comprehensive and feature-rich user management system that allows organizations to efficiently manage their digital ecosystem.

## What is user management

User management refers to the process of granting, configuring, and controlling access for users by administrators.
This encompasses the creation of user accounts, assigning roles and permissions, setting authentication methods, and managing user-related data.

## Availability

User management is available in all [[= product_name =]] versions.

## How does user management work

[[= product_name =]] simplifies user management with an intuitive and powerful system of accounts, Roles, Permissions, groups, and Segments.
You can find all user groups and Users in the **Admin Panel** by selecting **Users**.
Here, you can manage users, their relations, Roles, and Policies.

> ![User's section](users_section.png)

Reviewer: ```suggestion
![Users section](users_section.png "Users section")
```

Here's how it works:

- User accounts - create and manage user accounts. This includes capturing user information, such as name, email, and profile details.

- Roles and Permissions - define Roles and assign Permissions to them. This ensures that users have appropriate access to content and functionalities. Roles can be customized to match the organization's specific needs.

- Authentication methods - enable multiple authentication methods, including traditional username and password, OAuth, and external service logins. This flexibility allows organizations to adapt to various user authentication requirements.

- User segmentation - segment users based on criteria such as demographics, behavior, or preferences. This segmentation enables personalized content delivery and targeted marketing.

> - Invitations - invite users to join a platform streamlining an onboarding process, sending invitations for exclusive content or events.

Reviewer: ```suggestion
- Invitations - invite users to join a platform, to streamline an onboarding process, or send exclusive content or event invitations.
```

### User Roles and Permissions

[[= product_name_base =]] allows you to define custom user [Roles with granular Permissions](permission_overview.md), ensuring that users have access to only the specific parts of the system they need.
Furthermore, you can create user groups to simplify Permission management.
Assign multiple users to a group to ensure consistency and ease of access control.
This helps maintain effortless security and control.

To help you understand further the role each element serves, here's a brief summary: 

- Role - represents a collection of Permissions that can be assigned to users or user groups. Roles streamline permission management by grouping related Permissions together.
- Permission - defines a specific action or access level that can be granted or denied within the system.
- Policy - is a set of rules or conditions that determine under what circumstances a specific permission is granted or denied by applying Limitations.
Policies allow for fine-grained control of access based on various factors, such as user attributes or system states.

### Custom Policies

> [Tailor user access control](custom_policies.md) to your unique requirements using custom policies. Define complex rules and access criteria for different users or groups.

Reviewer: ```suggestion
[Tailor user access control](custom_policies.md) to your unique requirements by creating custom policies. Define complex rules and access criteria for different users or groups.
```

- Invitations — invite users to join a platform streamlining an onboarding process, sending invitations for exclusive content or events.

- Customer groups — organize users into customer groups, which helps in delivering tailored experiences and content to specific segments.

![User management](user_management.png)

## Capabilities

The detailed capabilities of [[= product_name_base =]] User Management,
which provide organizations with the tools they need to deliver personalized, secure,
and efficient user experiences while ensuring that user access and content delivery align with their business goals and strategies.

### User Roles and Permissions

[[= product_name_base =]] allows you to define custom user [Roles with granular Permissions](permission_overview.md), ensuring that users have access to only the specific parts of the system they need.
Furthermore, you can create user groups to simplify Permission management.
Assign multiple users to a group to ensure consistency and ease of access control.
This helps maintain effortless security and control.

> **Permission:**

Reviewer 2: Just an option to consider - list is more clear (in my opinion), and additionally, in the current form, when each sentence begins with "A" letter ('A Permission...', 'A Role...'), I see it as a letter numbered list.

Option:

- **Permission** - defines a specific action or access level that can be granted or denied within the system.

- **Role** - represents a collection of Permissions that can be assigned to users or user groups.
Roles streamline permission management by grouping related Permissions together.

- **Policy** - is a set of rules or conditions that determine under what circumstances a specific permission is granted or denied by applying Limitations.
Policies allow for fine-grained control of access based on various factors, such as user attributes or system states.

You can find all user groups and Users in the **Admin Panel** by selecting **Users**. 
Here, you can manage users, their relations, Roles and Policies.

![User's section](users_section.png)

Here's how it works:

- user accounts - create and manage user accounts. This includes capturing user information, such as name, email, and profile details.

- Roles and Permissions - define Roles and assign Permissions to them. This ensures that users have appropriate access to content and functionalities. Roles can be customized to match the organization's specific needs.

- authentication methods - enable multiple authentication methods, including traditional username and password, OAuth, and external service logins. This flexibility allows organizations to adapt to various user authentication requirements.

- user segmentation - Segment users based on criteria such as demographics, behavior, or preferences. This segmentation enables personalized content delivery and targeted marketing.

- invitations - invite users to join a platform. This is useful for onboarding new users or sending invitations for exclusive content or events.

> - customer groups - organize users into customer groups, which helps in delivering tailored experiences and content to specific segments.

Reviewer 3: How about to begin with capital letter?

Policies allow for fine-grained control of access based on various factors, such as user attributes or system states.

### Custom Policies

[Tailor user access control](custom_policies.md) to your unique requirements using custom policies. Define complex rules and access criteria for different users or groups.

### Limitations

[Implement Limitations](limitations.md) on user actions based on specific criteria, such as time-based restrictions or geographic locations.

### Authentication methods

[[= product_name_base =]] offers flexibility in authentication methods to cater to different user bases and security requirements.

The system's OAuth integration allows users to log in with their existing social media or third-party credentials, supporting OAuth 2.0 for easy integration with services like Google, Facebook, and Twitter.

![Log in via Google](log_in_via_google.png)

Available options:

- [Username and password](user_management.md) - ideal for most users, this traditional method offers a secure login process with username and password.

> - [OAuth](oauth_authentication.md) - integrating OAuth authentication allows users to log in using their existing social media or third-party credentials, streamlining the registration process.

Reviewer 3: I would rewrite it a little bit as it sounds like repetition from line: 82.

- [external service](add_login_through_external_service.md) - for enterprise users, integration with an external service like Active Directory or LDAP ensures that user authentication aligns with the organization's existing systems.

### Invitations

The [invitation system](invitations.md) streamlines user onboarding and engagement.
Track the status of invitations, including when they were sent, whether they were accepted, and the actions taken by users who accepted them.

![Invitations](users_invitation.png)

### User segmentation and personalization

[[= product_name_base =]]'s segmentation and personalization features allow organizations to deliver customized user experiences.

> Track user behavior, such as page views, search queries, and interactions, to create Segments and Segment Groups for users who share similar actions.

Reviewer 3: ```suggestion
Track user behavior, such as page views, search queries, and interactions, to create Segments and Segment Groups for users who share similar behaviors.
```

![Segment Groups](../administration/img/admin_panel_segment_groups.png)

Possible uses:

- Demographics - Segment users based on demographic data such as age, location, and gender to personalize content, promotions, and recommendations.
- Behavior - Tailor content based on user behavior, such as frequent content consumption, shopping patterns, or search history, ensuring users see what they're interested in.

> - Preferences - Utilize user preferences to offer a customized experience, from language preferences to Content Type preferences.

Reviewer 3: Begin with capital letter, after hyphen, lower case.

### Customer groups

The customer group functionality allows for targeted content delivery and service offerings.

> Set specific permissions for customer groups to control who can access and edit certain content ot get specific recommendations.

Reviewer 3: ```suggestion
Set specific permissions for customer groups to control who can access and edit certain content to get respective recommendations.
```

Possible uses:

- product recommendations - Create customer groups based on product preferences and offer tailored product recommendations.

> - targeted marketing campaigns - Segment users by demographics or behavior and deliver targeted marketing campaigns that resonate with specific groups.

Reviewer 3: Rewrite as it resembles lines: 108.

> - targeted marketing campaigns - Segment users by demographics or behavior and deliver targeted marketing campaigns that resonate with specific groups.

Reviewer 3: Capital letter -> - -> lower case.
